### ISSUE TITLE
ADS-Audit - no option to disable the retention - Effectv/Dell

### ISSUE THREAD

**Issue by raniangamuthu** [P3-Core-Dev-Team/P3-Q-A#100]
**Describe the bug**
Max is 720 days for retention but no option to disable the retention if the user prefers not to apply retention for the audit log

---

**To reproduce**
System Management >> Setting >>Audit Retention Period
---

**Screenshots**
![Screenshot 2024-03-25 084458](https://github.com/P3-Core-Dev-Team/P3-Q-A/assets/69497085/94993658-4e23-416f-ba99-75f459324d37)

---

Additional context
Build no : ADS v
Client : Effectv

**Comment by kilam005** [comment]
Audit logs are no longer maintained after 2 years.
Storing data for long days will encounter storage cost and the purpose of audit is to track for some fail over or for security reasons. This will have to be checked almost immediately once the incident happens.

upto 2 years is a long durations.

I have been asking the team to provision some thing like an audit archive application to store audit longs older than the define audit retention duration. There is a idea for this but its not converted into a requirement yet.

**Comment by kilam005** [comment]
@maheswarareddyi please get this planned with saumya and team

**Comment by kilam005** [comment]
@raniangamuthu 
please provide all expectations as a requirment hint list to get the team focus on the requirment drafting first.

**Comment by maheswarareddyi** [comment]
@raniangamuthu  As discussed on April 12, 2024, Please consolidate the expectations and provide clarifications on the questions asked during the discussion to frame the requirements more clearly.

**Comment by kilam005** [comment]
Requriment drafting is in progress and expected to move to next phase by eod of next week.

**Comment by maheswarareddyi** [comment]
The requirement document has been submitted for review and is expected to be signed off by May 31, 2024.

**Comment by maheswarareddyi** [comment]
A walkthrough of the requirement changes was conducted with the stakeholders. All concerns raised and comments provided were addressed and resolved. The requirement sign-off is expected by June 13, 2024.

**Comment by maheswarareddyi** [comment]
Requirements are signed off by the stakeholders and delivered to the development team.

@selvakumarchandran @prabhakaranUI Please find the signed-off requirements [here](https://platform3solutionsllc.sharepoint.com/:f:/r/Shared%20Documents/02.%20PRODUCT/Archon%20Data%20Platform/Requirements/ADS_REQ_0039_Audit%20Application?csf=1&web=1&e=oyYdVk)

**Comment by kilam005** [comment]
Planning review was completed and development has started.

**Comment by kilam005** [comment]
Work is in progress. ETA expcted by 5th July. Actual status need to be confirmed.

@selvakumarchandran or @ShaliniSettu please update

**Comment by ShaliniSettu** [comment]
audit tracker application feature development is in-progress

**Comment by kilam005** [comment]
@ShaliniSettu what is the ETA

**Comment by ShaliniSettu** [comment]
@kilam005 development completed, dev testing is going on. It is expected to deliver by tomorrow EOD

**Comment by ShaliniSettu** [comment]
this fix will be deliver along with audit application feature

**Comment by Harik726** [comment]
Development has been completed.
Will be released in next re version

**Comment by kilam005** [comment]
Fix is under reveiw.
@ashoksri0 need to approve to package it.
@ashoksri0 has confirmed to start review today.

**Comment by maheswarareddyi** [comment]
PR review is in progress and we expected to get the review done by @ashoksri0 this week.

**Comment by kilam005** [comment]
@ashoksri0  PR review pending for about 2 weeks now. Is it completed? please inform ETA to pass this for testing

**Comment by kilam005** [comment]
PR has been merged and will be send for testing by EOW

**Comment by Udhayanila8** [comment]
@kilam005 As confirmed with @Harik726, the build has still not been delivered for testing.

**Comment by Harik726** [comment]
@kilam005  and @Udhayanila8  Feature have been completed. Will be delivered in next release

**Comment by kilam005** [comment]
Delivery has been posponted from the planned date. It is scheudled to be deployed today and handed over for Testing.

**Comment by kilam005** [comment]
Testing started last week. Halted coz of application installation flow issue. Dev team has fixed and given correction and handed over for testing again today.

**Comment by Udhayanila8** [comment]
http://192.168.56.146:4081/
version: 2024.10.45564_RC3

Basic scenarios are working fine, but still testing for Audit Application feature is InProgress. Once testing is done will give closure to this issue.

Screenshots: 
![image](https://github.com/user-attachments/assets/3ae660aa-e0c7-4db6-9e17-65facb98e8c6)
![image](https://github.com/user-attachments/assets/9a338774-246a-4da6-93c5-73ec7ebe053c)
![image](https://github.com/user-attachments/assets/7d59bc07-7458-4155-9872-d4a0da551ed9)
![image](https://github.com/user-attachments/assets/6406e269-ab88-47da-9fe5-4cdd72872b3b)
![image](https://github.com/user-attachments/assets/8f249e02-88f7-48d0-9638-155da5974806)
![image](https://github.com/user-attachments/assets/5e0fdc39-7362-4d04-9665-4474717b7857)

**Comment by kilam005** [comment]
@Udhayanila8 please start verfying on the new build shared today.

**Comment by maheswarareddyi** [comment]
Feature testing is in progress by @Udhayanila8 who identified some blockers, once the blockers are fixed and delivered this feature can be tested and certified by the testing team.

**Comment by kilam005** [comment]
Previously identfiied blockers were relsoved and new build was shared. @Udhayanila8 to continue testing.

**Comment by maheswarareddyi** [comment]
Feature testing is in progress by @Udhayanila8.

**Comment by Srividhya2506** [comment]
For the audit application, the test case execution was completed. 3 issues need to be retest. For 2 issues, comments have been added, and we are waiting for the response; another issue is in IR status. Once the 3 issues got retested, we will add verification tag and added issues list for reference.
https://github.com/P3-Core-Dev-Team/ADS_BE/issues/3392
https://github.com/P3-Core-Dev-Team/ADS_BE/issues/3375
https://github.com/P3-Core-Dev-Team/ADS_BE/issues/3382

**Comment by ShaliniSettu** [comment]
Updates for the issues mentioned by @Srividhya2506

[Issue #3392](https://github.com/P3-Core-Dev-Team/ADS_BE/issues/3392) – PR has been merged and will be available in next release of 2024.10.x.
[Issue #3375](https://github.com/P3-Core-Dev-Team/ADS_BE/issues/3375) – Invalid issue. (@Srividhya2506 said that she will be discussing with @kilam005 on this)
[Issue #3382](https://github.com/P3-Core-Dev-Team/ADS_BE/issues/3382) – Retest of this issue has been blocked due to [Issue #3370](https://github.com/P3-Core-Dev-Team/ADS_BE/issues/3370), and the pr for this issue has been merged and will be available in 2024.10.x.

**Comment by kilam005** [comment]
@Udhayanila8 This will be avaialble for Testing tomorrow.

**Comment by Thilak-Saravanan** [comment]
@Udhayanila8 Above mentioned issues are available for retesting, Kindly revalidate.

**Comment by Srividhya2506** [comment]
This issue https://github.com/P3-Core-Dev-Team/ADS_BE/issues/3370 is still in RCA.
Retest of issue https://github.com/P3-Core-Dev-Team/ADS_BE/issues/3382 has been blocked due to https://github.com/P3-Core-Dev-Team/ADS_BE/issues/3370. As of now, https://github.com/P3-Core-Dev-Team/ADS_BE/issues/3392 This issue will be retested. Once fix is provided for mentioned issues, verified will be given by testing team.

**Comment by kilam005** [comment]
@ShaliniSettu  and @selvakumarchandran 
Check and give full closure.

**Comment by Srividhya2506** [comment]
http://192.168.56.146:4081/ and https://testregion.platform3solutions.com/applications
2024.10.45564_RC11

Installed Audit Tracker Application, stored audit records through archive audits job. Able to perform prebuild searches, user created searches and can able to access prebuild table. Performed compliance and job management related jobs. Received alert email when audit tracker application is not installed, and audit records are ready for dispose. Working as expected in both 146 server and serverless region.

Screenshots:
![Screenshot_349](https://github.com/user-attachments/assets/e5577f5a-feb6-41a2-9655-e4acdaffe368)
![Screenshot_350](https://github.com/user-attachments/assets/99164cd9-5b50-494e-bb69-2c2a3f2408e8)
![Screenshot_351](https://github.com/user-attachments/assets/a81ccf0a-ff73-4e05-9c44-5034822300b7)
![Screenshot_352](https://github.com/user-attachments/assets/903d1ac0-0cfc-47dc-a799-85ff9434be37)
![Screenshot_353](https://github.com/user-attachments/assets/73ee1d70-7d17-44a4-87ca-792efdae9c65)
![Screenshot_354](https://github.com/user-attachments/assets/7a6a63da-a280-45c3-85fd-f661a2a8b578)
![Screenshot_355](https://github.com/user-attachments/assets/465d7e1a-73b4-49de-88ce-3ce254298999)
![Screenshot_358](https://github.com/user-attachments/assets/67f2b8e7-55b3-45a0-ad35-f205545c9694)
![Screenshot_359](https://github.com/user-attachments/assets/03a48d0c-21b0-4212-b120-99ab7bc3aca5)
![Screenshot_360](https://github.com/user-attachments/assets/74dbc4b9-99dd-4ffa-b57e-b653ff68fa0e)
Serverless
![image](https://github.com/user-attachments/assets/97d80431-8b55-4566-90d0-9fd2e950bfa3)
![image](https://github.com/user-attachments/assets/36bee3f0-15ed-44fb-b021-cfa36a5691f6)
![image](https://github.com/user-attachments/assets/ba369e6c-d4e6-46bc-bcf4-6c46b6e258b4)

Document:
https://platform3solutionsllc.sharepoint.com/:f:/r/Shared%20Documents/02.%20PRODUCT/TESTING/PRODUCT%20Test%20Report/ADS/Test%20Reports/New%20Feature%20Testing/Audit%20Application%20Test%20Reports?csf=1&web=1&e=gm0NJC

**Comment by Thilak-Saravanan** [comment]
Available in 2025.01.x version release.

**Comment by Thilak-Saravanan** [comment]
@raniangamuthu 

This feature was released in 2025.01.45679 and it is available in UAT(Pre-prod region).
Kindly validate and provide closure

**Comment by raniangamuthu** [comment]
Team, I do not see an option to disable the audit retention period.
Guide me how to validate.
![Image](https://github.com/user-attachments/assets/3f44c58b-9e0a-42b8-ba97-bb299066d0eb)

**Comment by maheswarareddyi** [comment]
@raniangamuthu 
As previously communicated, we have not provided an option to completely disable retention, as storing logs for an extended period can impact database performance (Please refer to Malik's comment dated March 26, 2024).

To address this, we offer a prebuilt audit tracker application that allows users to archive and store audit logs, perform search lookups, and run compliance workflows. Users can manually archive audit logs using the Archive Audits job, or the system will automatically archive logs during the daily audit cleanup process. During this process, logs that are ready for disposal will be archived into the audit tracker application before being removed.

For more information, please refer to the requirements [here](https://platform3solutionsllc.sharepoint.com/:f:/r/Shared%20Documents/02.%20PRODUCT/Archon%20Data%20Platform/Requirements/ADS_REQ_0039_Audit%20Application?csf=1&web=1&e=K0YM5g) and the user guide [here](https://platform3solutionsllc.sharepoint.com/:b:/r/Shared%20Documents/02.%20PRODUCT/Product%20Releases/Archon%20Data%20Store/v2025.01/v2025.01.45679/Release%20Package/Archon%20Data%20Store%20v2025.01.45679/Release%20Support%20Documents/Audit%20Application%20-%20User%20Guide.pdf?csf=1&web=1&e=ews1Ky).

**Comment by raniangamuthu** [comment]
thanks for adding this feature
